src/test/java/org/jenkinsci/plugins/workflow/cps/PersistenceProblemsTest.java

```diff
@@ -313,7 +312,11 @@ public void inProgressButProgramLoadFailure() throws Exception {
         story.thenWithHardShutdown( j -> {
             WorkflowRun run = runBasicPauseOnInput(j, DEFAULT_JOBNAME, build);
             CpsFlowExecution cpsExec = (CpsFlowExecution)(run.getExecution());
-            cpsExec.getProgramDataFile().delete();
+            // Wait until program.dat is written and then delete it.
+            while (!Files.exists(cpsExec.getProgramDataFile().toPath())) {
+                Thread.sleep(100);
+            }
+            Files.delete(cpsExec.getProgramDataFile().toPath());
```

Comment:
Previously, the call to `delete` was sometimes running before `program.dat` had been written to disk, so the test would fail because the Pipeline resumed normally after the restart and just hung at an `input` step.